Determine Release body cannot accept certain characters

Apparently we can't use "`" in the release body.  I should add logic to catch instances that could screw up the pipeline like this.
#### Description
The `PR_BODY` variable in the workflow is intended to store the text for the pull request body. However, if the text includes special characters (e.g., backticks, quotes, or other shell-sensitive characters), it can cause unexpected behavior or script errors. For instance, backticks in the string are being interpreted as command substitution, leading to errors like `command not found`.

#### Example of the Issue
An example where backticks are present in the `PR_BODY`:
```bash
PR_BODY="When the primary unit is a frequency and the secondary unit is not \`null\`, we flip-flop the units for uncertainty lookup, and treat frequency as the secondary unit.\n"
```
This results in:
```
/home/runner/work/_temp/a15d0079-43d3-4b07-8f58-6306fea05645.sh: line 1: null: command not found
```

#### Expected Behavior
Special characters in the `PR_BODY` should be properly escaped or handled to ensure they are treated as part of the string and not as commands or control characters.

#### Proposed Solution
1. Use a reliable escape mechanism to handle special characters in `PR_BODY`.
2. Wrap the variable in single quotes (`'`) or escape problematic characters (e.g., backticks, quotes).
3. Optionally, use tools like `printf` or `jq` for robust handling of multi-line strings or JSON-like structures.

#### Suggested Code Fix
In the workflow:
```bash
PR_BODY=$(echo "$PR_BODY" | sed 's/`/\\`/g' | sed 's/"/\\"/g')
```

Alternatively, use `jq` for escaping:
```bash
PR_BODY=$(echo "$PR_BODY" | jq -R .)
```

#### Additional Considerations
- Test for other special characters like `$`, `\`, and `&` to ensure they are also escaped properly.
- Ensure compatibility with GitHub Actions' environment and syntax for setting output variables.

### Another example:
```
Determine Release Body
Run PR_BODY="Allow `A` and `V` as primary unit types.\r\nAuto-calculate polarity and remove user input.\r\nRefactor `has_range[1](https://github.com/Johnson-Gage-Inspection-Inc/xl-Electrical-Workbook/actions/runs/14138669787/job/39616130538#step:6:1)_only` out completely from VBA code.\n"
/home/runner/work/_temp/30cafb3e-93e9-4895-af76-a17cc0d046ec.sh: line 1: A: command not found
/home/runner/work/_temp/30cafb3e-93e9-4895-af76-a17cc0d046ec.sh: line 1: V: command not found
/home/runner/work/_temp/30cafb3e-93e9-4895-af76-a17cc0d046ec.sh: line 1: has_range1_only: command not found
Error: Process completed with exit code 127.
```

--- a/.github/workflows/upload-release.yml
+++ b/.github/workflows/upload-release.yml
@@ -290,8 +290,15 @@ jobs:
               echo "⚠️ PR found, but no description available. Using commit message."
               PR_BODY=""
             fi
+            
+            # Enhanced escaping: Replace backticks and other special characters
+            # Use sed to replace backticks with their HTML entity equivalent
+            PR_BODY=$(echo "$PR_BODY" | sed 's/`/\\`/g' | sed 's/"/\\"/g')
+            
+            # Then use jq for additional safety in JSON escaping
             PR_BODY_ESCAPED=$(echo "$PR_BODY" | jq -Rs '.')
-            echo "✅ PR Body Retrieved."
+            
+            echo "✅ PR Body Retrieved and Properly Escaped."
             echo "PR_BODY=$PR_BODY_ESCAPED" >> "$GITHUB_OUTPUT"
           fi
         env:
@@ -301,7 +308,11 @@ jobs:
         id: commit_message
         run: |
           COMMIT_MSG=$(git log -1 --pretty=%B)
+          
+          # Apply the same escaping to commit messages
+          COMMIT_MSG=$(echo "$COMMIT_MSG" | sed 's/`/\\`/g' | sed 's/"/\\"/g')
           COMMIT_MSG_ESCAPED=$(echo "$COMMIT_MSG" | jq -Rs '.')
+          
           echo "COMMIT_MSG=$COMMIT_MSG_ESCAPED" >> "$GITHUB_OUTPUT"
 
       - name: Determine Release Body
@@ -317,7 +328,10 @@ jobs:
             RELEASE_BODY="$PR_BODY"
           fi
 
-          # Ensure proper escaping for GitHub Actions output
+          # Additional safety check to ensure no unescaped backticks or special characters remain
+          RELEASE_BODY=$(echo "$RELEASE_BODY" | sed 's/^"//;s/"$//' | sed 's/\\"/"/g')
+          
+          # Use heredoc to avoid shell interpretation issues
           {
             echo "RELEASE_BODY<<EOF"
             echo "$RELEASE_BODY"

